lib/node_modules/@stdlib/lapack/base/dptts2/test/test.ndarray.js

```diff
@@ -0,0 +1,148 @@
+var abs = require( '@stdlib/math/base/special/abs' );
+var EPS = require( '@stdlib/constants/float64/eps' );
+var dptts2 = require( './../lib/ndarray.js' );
+
+
+// FUNCTIONS //
+
+/**
+* Tests for element-wise approximate equality.
+*
+* @private
+* @param {Object} t - test object
+* @param {Collection} actual - actual values
+* @param {Collection} expected - expected values
+* @param {number} rtol - relative tolerance
+*/
+function isApprox( t, actual, expected, rtol ) {
+	var delta;
+	var tol;
+	var i;
+
+	t.strictEqual( actual.length, expected.length, 'returns expected value' );
+	for ( i = 0; i < expected.length; i++ ) {
+		if ( actual[ i ] === expected[ i ] ) {
+			t.strictEqual( actual[ i ], expected[ i ], 'returns expected value' );
+		} else {
+			delta = abs( actual[ i ] - expected[ i ] );
+			tol = rtol * EPS * abs( expected[ i ] );
+			t.ok( delta <= tol, 'within tolerance. actual: '+actual[ i ]+'. expected: '+expected[ i ]+'. delta: '+delta+'. tol: '+tol+'.' );
+		}
+	}
+}
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof dptts2, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function has an arity of 12', function test( t ) {
+	t.strictEqual( dptts2.length, 12, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function solves a tridiagonal system of the form `A * X = B` using `L * D * L^T` factorization of `A` (row-major, offsetD=3, offsetE=4, offsetB=6)', function test( t ) {
+	var expected;
+	var out;
+	var D;
+	var E;
+	var B;
+
+	D = new Float64Array( [ 0.0, 0.0, 0.0, 1.0, 1.0, 1.0 ] );
+	E = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 2.0, 3.0 ] );
+	B = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+
+	expected = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 11.0, 38.0, -5.0, -18.0, 2.0, 6.0 ] );
+	out = dptts2( 3, 2, D, 1, 3, E, 1, 4, B, 2, 1, 6 );
+	t.strictEqual( out, B, 'returns expected value' );
+	isApprox( t, B, expected, 1.0e-7 );
+
+	t.end();
+});
+
+tape( 'the function solves a tridiagonal system of the form `A * X = B` using `L * D * L^T` factorization of `A` (row-major, offsetD=3, strideD=2, offsetE=4, offsetB=6)', function test( t ) {
+	var expected;
+	var out;
+	var D;
+	var E;
+	var B;
+
+	D = new Float64Array( [ 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0 ] );
+	E = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 2.0, 3.0 ] );
+	B = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+
+	expected = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 11.0, 38.0, -5.0, -18.0, 2.0, 6.0 ] );
+	out = dptts2( 3, 2, D, 2, 3, E, 1, 4, B, 2, 1, 6 );
+	t.strictEqual( out, B, 'returns expected value' );
+	isApprox( t, B, expected, 1.0e-7 );
+
+	t.end();
+});
+
+tape( 'the function solves a tridiagonal system of the form `A * X = B` using `L * D * L^T` factorization of `A` (row-major, strideB1=-2, strideB2=-1)', function test( t ) {
```

Comment:
Here, you are showing that the function supports negative strides for `B`. Update the description accordingly. Etc.